File: rxbinding/src/main/java/com/jakewharton/rxbinding2/view/MenuItemActionViewEvent.java
package com.jakewharton.rxbinding2.view;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.MenuItem;

/**
 * An action view event on a menu item.
 * <p>
 * <strong>Warning:</strong> Instances keep a strong reference to the menu item. Operators that
 * cache instances have the potential to leak the associated {@link Context}.
 */
public abstract class MenuItemActionViewEvent {
  MenuItemActionViewEvent() {
  }

  /** The menu item from which this event occurred. */
  public abstract @NonNull MenuItem menuItem();
}

File: rxbinding/src/main/java/com/jakewharton/rxbinding4/view/MenuItemActionViewEvent.kt
package com.jakewharton.rxbinding4.view

import android.content.Context
import android.view.MenuItem

/**
 * An action view event on a menu item.
 *
 * **Warning:** Instances keep a strong reference to the menu item. Operators that
 * cache instances have the potential to leak the associated [Context].
 */
sealed class MenuItemActionViewEvent {
  /** The menu item from which this event occurred.  */
  abstract val menuItem: MenuItem
}

data class MenuItemActionViewCollapseEvent(
  override val menuItem: MenuItem
) : MenuItemActionViewEvent()

data class MenuItemActionViewExpandEvent(
  override val menuItem: MenuItem
) : MenuItemActionViewEvent()
